eng/pipelines/templates/jobs/ci.yml

```diff
@@ -47,301 +47,82 @@ parameters:
     default: false
 
 jobs:
-  - job: Build
-    pool:
-      name: $(WINDOWSPOOL)
-      image: $(WINDOWSVMIMAGE)
-      os: windows
-
-    variables:
-      Codeql.Enabled: true
-      Codeql.BuildIdentifier: ${{ parameters.ServiceDirectory }}
-      Codeql.SkipTaskAutoInjection: false
-      SDKType: ${{ parameters.SDKType }}
-
-    steps:
-      - ${{ if not(contains(variables['Build.DefinitionName'], '-pr - ')) }}:
-        - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
-          parameters:
-            Paths:
-              - "/*"
-              - "!SessionRecords"
-      - template: /eng/pipelines/templates/steps/install-dotnet.yml
-      - template: /eng/common/pipelines/templates/steps/set-test-pipeline-version.yml
-        parameters:
-          PackageName: "Azure.Template"
-          ServiceDirectory: "template"
+  - ${{ if eq(parameters.ServiceDirectory, 'auto') }}:
```

Comment:
Is there any reason to condition here? Should we just create a simple 1 job matrix for the non-auto case? So, we only have to condition the one little part?

Reply:
The diff looked scarier than it actually was and so what you have seems reasonable.